File: easy/283.Move Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int curr=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]!=0)swap(nums[curr++],nums[i]);
            // for(int j=0;j<nums.size();j++){
            //     if(nums[j]!=0) swap(nums[curr++],nums[j]);
            // }
        }
    }
};

File: easy/283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        nonz=0
        for i in range(len(nums)):
            if nums[i]!=0:
                nums[i],nums[nonz]=nums[nonz],nums[i]
                nonz+=1
        
